server/src/main/java/org/elasticsearch/env/NodeMetadata.java

```diff
@@ -113,7 +113,7 @@ public void verifyUpgradeToCurrentVersion() {
         assert (nodeVersion.equals(Version.V_EMPTY) == false) || (Version.CURRENT.major <= Version.V_7_0_0.major + 1)
             : "version is required in the node metadata from v9 onwards";
 
-        if (nodeVersion.before(Version.CURRENT.minimumCompatibilityVersion())) {
+        if (Build.isWireCompatibleWithCurrentAllowingLegacy(nodeVersion.toString()) == false) {
             throw new IllegalStateException(
```

Comment:
There's a separate ticket for removing `Version` from `NodeMetadata`.

I think it makes sense to move the call to `Version` into `Build` so we can re-work how we do that sort of comparison in one place rather than in a bunch of places in the codebase.

I would also be okay leaving this as-is and handling it when we work on `NodeMetadata`.